contracts/EscrowAndStaking/StakingThales.sol

```diff
@@ -124,151 +136,183 @@ contract StakingThales is IStakingThales, Initializable, ProxyOwned, ProxyReentr
+    /// @return the total rewards claimed for the account until now
     function getAlreadyClaimedRewards(address account) external view returns (uint) {
         return stakerLifetimeRewardsClaimed[account];
     }
 
+    /// @notice Get the total fees claimed for the account until now
+    /// @param account to get the total fees claimed for
+    /// @return the total fees claimed for the account until now
     function getAlreadyClaimedFees(address account) external view returns (uint) {
         return stakerFeesClaimed[account];
     }
 
+    /// @notice Get the rewards funds available on the rewards pool
+    /// @return the rewards funds available on the rewards pool
     function getContractRewardFunds() external view returns (uint) {
         return stakingToken.balanceOf(address(ThalesStakingRewardsPool));
     }
 
+    /// @notice Get the fee funds available on the staking contract
+    /// @return the fee funds available on the staking contract
     function getContractFeeFunds() external view returns (uint) {
         return feeToken.balanceOf(address(this));
     }
 
-    function setDistributeFeesEnabled(bool _distributeFeesEnabled) external onlyOwner {
+    /// @notice Set staking parametars
+    /// @param _claimEnabled enable/disable claim rewards
+    /// @param _distributeFeesEnabled enable/disable fees distribution
+    /// @param _durationPeriod duration of the staking period
+    /// @param _unstakeDurationPeriod duration of the unstaking cooldown period
+    function setStakingParameters(
+        bool _claimEnabled,
+        bool _distributeFeesEnabled,
+        uint _durationPeriod,
+        uint _unstakeDurationPeriod
+    ) external onlyOwner {
+        claimEnabled = _claimEnabled;
         distributeFeesEnabled = _distributeFeesEnabled;
-        emit DistributeFeesEnabled(_distributeFeesEnabled);
-    }
-
-    function setFixedPeriodReward(uint _fixedReward) external onlyOwner {
-        fixedPeriodReward = _fixedReward;
-        emit FixedPeriodRewardChanged(_fixedReward);
-    }
-
-    function setPeriodExtraReward(uint _extraReward) external onlyOwner {
-        periodExtraReward = _extraReward;
-        emit PeriodExtraRewardChanged(_extraReward);
-    }
+        durationPeriod = _durationPeriod;
+        unstakeDurationPeriod = _unstakeDurationPeriod;
 
-    function setClaimEnabled(bool _claimEnabled) external onlyOwner {
-        claimEnabled = _claimEnabled;
         emit ClaimEnabled(_claimEnabled);
-    }
-
-    function setDurationPeriod(uint _durationPeriod) external onlyOwner {
-        durationPeriod = _durationPeriod;
+        emit DistributeFeesEnabled(_distributeFeesEnabled);
         emit DurationPeriodChanged(_durationPeriod);
-    }
-
-    function setUnstakeDurationPeriod(uint _unstakeDurationPeriod) external onlyOwner {
-        unstakeDurationPeriod = _unstakeDurationPeriod;
         emit UnstakeDurationPeriodChanged(_unstakeDurationPeriod);
```

Comment:
if we join the setter, we better emit a single event

Reply:
Yes, I just checked are we using these events in the subgraph. We are ok, updated.